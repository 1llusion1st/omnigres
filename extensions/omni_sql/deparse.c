// clang-format off
#include <postgres.h>
#include <fmgr.h>
// clang-format on

#include "lib/stringinfo.h"
#include "nodes/nodes.h"
#include "nodes/parsenodes.h"

void omni_sql_deparseRawStmt(StringInfo str, RawStmt *raw_stmt);

char *omni_sql_deparse_statement(List *stmts) {
  char *result;
  StringInfoData str;
  ListCell *lc;

  initStringInfo(&str);

  foreach (lc, stmts) {
    omni_sql_deparseRawStmt(&str, castNode(RawStmt, lfirst(lc)));
    if (lnext(stmts, lc))
      appendStringInfoString(&str, "; ");
  }
  result = strdup(str.data);

  return result;
}
