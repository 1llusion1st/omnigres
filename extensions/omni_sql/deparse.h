#ifndef OMNI_SQL_DEPARSE_H
#define OMNI_SQL_DEPARSE_H

// clang-format off
#include <postgres.h>
#include <fmgr.h>
// clang-format on

char *omni_sql_deparse_statement(List *stmts);
#endif // OMNI_SQL_DEPARSE_H