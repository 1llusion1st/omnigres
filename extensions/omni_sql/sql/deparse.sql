select 'select * from t where v != 1'::omni_sql.statement;