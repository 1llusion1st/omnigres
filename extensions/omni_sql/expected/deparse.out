select 'select * from t where v != 1'::omni_sql.statement;
          statement           
------------------------------
 SELECT * FROM t WHERE v <> 1
(1 row)

